pos_container_deposit/static/src/js/models.js

```diff
@@ -40,7 +40,14 @@ odoo.define("pos_container_deposit.models", function (require) {
                  * When adding a product with container deposit, add its container deposit product
                  **/
                 super.add_orderline(...arguments);
-                if (line.container_deposit_product && !line.container_deposit_line) {
+                // Get unpayed orders orders from db.pos
+                const all_unpaid = this.pos.get_order_list().map((x) => x.cid);
+                const is_unpaid = all_unpaid.includes(line.order.cid);
```

Comment:
This looks like it is a good criterium, I tested on Runboat and it seems like this is indeed a measure of which orders are open in the current session and not paid yet.